### ISSUE TITLE
CNN images classification how to add “others” / “no classes detected” class?

### ISSUE THREAD

**Issue by randaller** [keras-team/keras#9066]
I am doing some classifier using convolutional neural network. For example, I have a webcam watching on my street, and trained on BMW 750 and Mercedes ML320 images classifier, so last softmax CNN layer shows me correct percentage of guessed "is this 750 or ml320 driving on my street?". But I don't wish to see [49%,51%] when both car models are not visible. I want to add 3rd class, "others", to have all the KIAs, Hondas, VWs etc to be softmax'ed 3rd class output of CNN. So I have created third class, trained on images of empty street, and images of street with buses, kias, hondas, and all others, and this approach is working and working well - now I am getting correct predictions from CNN, if there's objects of my interest (750 or 320) visible at the street or not (others class [99.8%,0.01%,0.01%] when no 750 or 320 visible). Is that correct approach when you need to get "no object detected" classifier output, or how to do this right?

```
model = Sequential()
model.add(Conv2D(32, (3, 3), input_shape=input_shape))
model.add(Activation('relu'))
model.add(MaxPooling2D(pool_size=(2, 2)))

model.add(Conv2D(32, (3, 3)))
model.add(Activation('relu'))
model.add(MaxPooling2D(pool_size=(2, 2)))

model.add(Conv2D(64, (3, 3)))
model.add(Activation('relu'))
model.add(MaxPooling2D(pool_size=(2, 2)))

model.add(Flatten())
model.add(Dense(64))
model.add(Activation('relu'))
model.add(Dropout(0.5))
model.add(Dense(3))
model.add(Activation('softmax'))

model.compile(loss='categorical_crossentropy', optimizer='adam', metrics=['accuracy'])
```

**Comment by maderafunk** [comment]
Sounds good. You could also think about creating 4 classes. 2 for BMW and Mercedes, one for empty frame, and one for other brands. 

You can use visualisations such as attention maps (https://github.com/raghakot/keras-vis) to check what area of the image the network is actually focusing on. Sometimes you might get good predictions, although the network is just making decisions based on totally different than expected features, such as the background.

**Comment by randaller** [comment]
Unfortunately I can't add empty street, as it is street - there's always from 2 to 20 cars parked (no 320/750's), so I'm using only 3rd class with "non-750 and non-320" images. Thanks, I will surely try to visualize cnn layers to check what does it detects. Any other ideas on how to improve "not-a-class" class?
